Simplenote/TagListViewController.m

```diff
@@ -473,30 +481,33 @@ - (void)menuDidClose:(NSMenu *)menu
 
 - (void)controlTextDidBeginEditing:(NSNotification *)notification
 {
-    NSTextView *textView = [notification.userInfo objectForKey:@"NSFieldEditor"];
-    self.tagNameBeingEdited = [textView.string copy];
+    self.tagNameBeingEdited = [notification.fieldEditor.string copy];
 }
 
 - (void)controlTextDidEndEditing:(NSNotification *)notification
 {
-    if (self.tagNameBeingEdited) {
-        // This can get triggered before renaming has started; don't do anything in that case
+    // This can get triggered before renaming has started; don't do anything in that case
+    if (!self.tagNameBeingEdited) {
+        return;
+    }
 
-        NSTextView *textView = [notification.userInfo objectForKey:@"NSFieldEditor"];
-        
-        [textView setSelectedRange:NSMakeRange(0, 0)]; // force de-selection of text
-        
-        // Note:
-        // Send a *COPY* of the string. Otherwise the internal string will be exposed, and this may lead to
-        // weird side effects.
-        NSString *newTagName = [textView.string copy];
-        
-        BOOL tagAlreadyExists = [self tagWithName:newTagName] != nil;
-        if ([newTagName length] > 0 && !tagAlreadyExists && ![self.tagNameBeingEdited isEqualToString:newTagName])
-            [self changeTagName:self.tagNameBeingEdited toName:newTagName];
-        
-        self.tagNameBeingEdited = nil;
+    // Force de-selection of text
+    NSTextView *textView = notification.fieldEditor;
+    [textView setSelectedRange:NSMakeRange(0, 0)];
+
+    // Send a *COPY* of the string. Otherwise the internal string will be exposed, and this may lead to
+    // weird side effects.
```

Comment:
like what? good comment, would love an example weird side effect

Reply:
DM'ing you a link to the original issue, since it's in the private repository!

Reply:
idea:

```
// Send a *COPY* of the string to avoid "NSLayoutManager: Loop / Crash"
```

Reply:
Sold!